app/Http/Middleware/TrimStrings.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace Rogue\Http\Middleware;
+
+use Illuminate\Foundation\Http\Middleware\TrimStrings as Middleware;
+
+class TrimStrings extends Middleware
+{
+    /**
+     * The names of the attributes that should not be trimmed.
+     *
+     * @var array
+     */
+    protected $except = [
+        //
+    ];
+}
```

Comment:
I'm confused about what this does, can you explain it to me?

Reply:
Yes! This property allows us to skip certain fields when trimming extra spaces from the end of strings (for example, Laravel normally includes `password` and `password_confirmation` here, since users might want to have passwords that end in a space character). Since we don't handle passwords in Rogue, I've left it blank for now.

Here's a [little blurb](https://laravel.com/docs/5.5/requests#input-trimming-and-normalization) from the official docs.

Reply:
Oh got it! Laravel has all the inner workings and this is just where we set the ones not to trim 👌

Reply:
Yep!